### ISSUE TITLE
How to get response body with `simple: false`?

### ISSUE THREAD

**Issue by borekb** [request/request-promise#174]
In plain `request` package, this code can be used to get both full request object as well as the body next to each other:

```js
req.get(url, {  }, (err, res, body) => {
    console.log('code: ' + res.statusCode);
    console.log('body: ' + body);
});
```

This works for all OK codes (2xx) as well as error codes. How to achieve the same with `request-promise`? After I call:

```ts
let response: RequestResponse = await rq.get(url, { resolveWithFullResponse: true, simple: false });
```

(`simple: false` is there because I wanted to prevent having to deal with some of my logic in the `catch` statement)  I'm not sure how to read from the `response` which is a subclass of `IncomingMessage`.

**Comment by analog-nico** [comment]
Hi @borekb you already got it right!

The `simple` option does not interfere with _what_ is passed to the promise. That is as you expect the sole job of the `resolveWithFullResponse`. If you set `resolveWithFullResponse: true` then you get the response returned.

``` js
try {
    let response = await rq.get(url, { resolveWithFullResponse: true, simple: true });
    console.log('code: ' + response.statusCode);
    console.log('body: ' + response.body);
} catch (err) {
    // Process the error
    // Equivalent to when you used Request with err !== null || res.statusCode !== 2xx
}
```

As you seem to use TypeScript there might be some issues with the type definitions. I never used them so I couldn't tell you if you have to look out for any eventual caveats in this regard.

**Comment by borekb** [comment]
_Sorry the example I've used contained `simple: true` while I meant `simple: false` as per the title; I've fixed the OP. But as you say, it should (and doesn't) matter._

Oh, there's `response.body`! I missed that in the examples and it's missing from [`request`'s type definition](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/request/index.d.ts#L144) so I was getting this:

![image](https://cloud.githubusercontent.com/assets/101152/22999694/2ada7a08-f3dd-11e6-814d-09fd21b66901.png)

Thanks for your answer!
